nautobot/utilities/utils.py

```diff
@@ -689,3 +720,220 @@ def pretty_str(self, node=None, depth=0):
 
     # Use pretty_str() as the string generator vs. just stringify the `Q` object.
     return pretty_str(query)
+
+
+def build_lookup_label(field_name, verbose_name):
+    """
+    Return lookup expr with its verbose name
+
+    Args:
+        field_name (str): Field name e.g slug__iew
+        verbose_name (str): The verbose name for the lookup exper which is suffixed to the field name e.g iew -> iendswith
+
+    Examples:
+        >>> build_lookup_label("slug__iew", "iendswith")
+        >>> "ends-with(iew)"
+    """
+
+    label = ""
+    search = re.search(r"(?<=__)\w+", field_name)
+    if search:
+        label = f"({search.group()})"
+
+    if FILTER_LOOKUP_MAP.get(verbose_name):
+        lookup_label = FILTER_LOOKUP_MAP[verbose_name]
+        verbose_name = "not-" + lookup_label if label.startswith("(n") else lookup_label
+
+    return verbose_name + label
+
+
+def get_all_lookup_exper_for_field(model, field_name):
```

Comment:
This looks like a typo and should be renamed:

```suggestion
def get_all_lookup_expr_for_field(model, field_name):
```